### PR TITLE
Fix the bug that warning message cannot be suppressed

### PR DESCRIPTION
**Pull request by Youri970410**
## Description
Fixed bug that always displayed warning Message when executing cmdlet based on MonitorCmdletBase.
#12889 

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Monitor/Monitor/MonitorCmdletBase.cs [hunk 1]
```diff
@@ -89,8 +89,12 @@ public override void ExecuteCmdlet()
 
             try
             {
-                WriteWarningWithTimestamp("*** The namespace for all the model classes will change from Microsoft.Azure.Management.Monitor.Management.Models to Microsoft.Azure.Management.Monitor.Models in future releases.");
-                WriteWarningWithTimestamp("*** The namespace for output classes will be uniform for all classes in future releases to make it independent of modifications in the model classes.");
+                var isWarning =  bool.Parse(System.Environment.GetEnvironmentVariable("SuppressAzurePowerShellBreakingChangeWarnings"));
```
Comment (dingmeng-xue) [inline src/Monitor/Monitor/MonitorCmdletBase.cs:92]: shouldSuppress is more accurate.

File: src/Monitor/Monitor/ChangeLog.md [hunk 2]
```diff
@@ -20,6 +20,9 @@
 
 ## Upcoming Release
 
+## Version 2.1.1
+* Fixed bug that always displayed warning Message when executing cmdlet based on MonitorCmdletBase [#12889]
```
Comment (dingmeng-xue) [inline src/Monitor/Monitor/ChangeLog.md:24]: My suggestion is 
```
Fixed the bug that warning message cannot be suppressed. [#12889]
```

File: src/Monitor/Monitor/MonitorCmdletBase.cs (hunk 1 above)
Comment (dingmeng-xue) [reply src/Monitor/Monitor/MonitorCmdletBase.cs:92]: What will happen if Environment is not set?

**Review by dingmeng-xue — CHANGES_REQUESTED** [review]

File: src/Monitor/Monitor/MonitorCmdletBase.cs (hunk 1 above)
Comment (dingmeng-xue) [reply src/Monitor/Monitor/MonitorCmdletBase.cs:92]: Test case is failed due to above reason.

**Review by dingmeng-xue — CHANGES_REQUESTED** [review]

File: src/Monitor/Monitor/MonitorCmdletBase.cs [hunk 3]
```diff
@@ -86,14 +86,22 @@ public override void ExecuteCmdlet()
             string reasonPhrase = null;
             string message = null;
             string exName = null;
-
+            string warningEnvironmentParam = "SuppressAzurePowerShellBreakingChangeWarnings";
```
Comment (dingmeng-xue) [inline src/Monitor/Monitor/MonitorCmdletBase.cs:89]: https://github.com/Azure/azure-powershell-common/blob/b58ba4053f9cbd2861813e6dc3bf4800713d1a7f/src/Common/CustomAttributes/BreakingChangeAttributeHelper.cs#L31

Should use this one.

**Review by dingmeng-xue — CHANGES_REQUESTED** [review]

File: src/Monitor/Monitor/MonitorCmdletBase.cs (hunk 1 above)
Comment (Youri970410) [reply src/Monitor/Monitor/MonitorCmdletBase.cs]: Got it,thanks

File: src/Monitor/Monitor/ChangeLog.md (hunk 2 above)
Comment (Youri970410) [reply src/Monitor/Monitor/ChangeLog.md]: Got it

File: src/Monitor/Monitor/MonitorCmdletBase.cs (hunk 3 above)
Comment (Youri970410) [reply src/Monitor/Monitor/MonitorCmdletBase.cs:89]: Got it,thanks.

**Review by dingmeng-xue — DISMISSED** [review]

**Review by dingmeng-xue — APPROVED** [review]

**Comment by cannehag** [comment]
When will this be released ?

**Comment by wyunchi-ms** [comment]
Hi @cannehag , this fix is already included in the latest release of Monitor. Please install the latest `Az.Monitor` to use it. Here is the way to install the latest release
```powershell
Install-Module Az.Monitor -Repository PSGallery -Force
```

**Comment by cannehag** [comment]
@wyunchi-ms I did update and still got the warnings. But after a new test it did work just fine. Thank you!

**Comment by wyunchi-ms** [comment]
Please try to set `SuppressAzurePowerShellBreakingChangeWarnings` to `True` in your environment.

**Comment by cannehag** [comment]
It is working! I hade to restart the PS session :)

**Pull request merged**